### PR TITLE
Fake teams

### PR DESCRIPTION
**Pull request by rorepo**
Changes made to gratipay/utils/fake_data.py to populate the new 'teams' table.

This is a work in progress. Intend to improve on some of the hardcoded values and also include subscriptions.

### CODE REVIEW

**Comment by whit537** [comment]
Thanks for the PR, @rorepo! Mind updating the failing test? Travis is :red_circle:.

**Comment by whit537** [comment]
> This is a work in progress.

Sorry, missed that comment in my excitement at a new (and mysterious! :-) contributor. I'll wait for the high sign from you before reviewing further. :-)

[!m](http://motivate.im/) @rorepo

**Comment by rorepo** [comment]
Improved teams fake data with a bit of randomisation and also updated fake data test script.

**Comment by rorepo** [comment]
I have added fake subscriptions and correspondingly updated the fake test script as well.

This is now ready for review.

File: gratipay/utils/fake_data.py [hunk 1]
```diff
@@ -237,13 +281,24 @@ def populate_db(db, num_participants=100, num_tips=200, num_teams=5, num_transfe
...
-        members = random.sample(participants, random.randint(1, 3))
-        for p in members:
-            team.add_member(p)
-        participants.append(team)
+    teams = []
+    teamowners = random.sample(participants, num_teams)
+    for teamowner in teamowners:
+        teams.append(fake_team(db, teamowner))
+
+    print("Making Subscriptions")
```
Comment (whit537) [inline gratipay/utils/fake_data.py]: Have we stopped making `tips` already in `fake_data`? (`tips` are what `subscriptions` replaced.) If not, let's do so on this PR.

[!m](http://motivate.im/) @rorepo

**Comment by rorepo** [comment]
@whit537 - not quite clear about the 'noisy commit'. Is this about the master branch having had other updates in between? 
Haven't used git before, so a bit unclear on concepts. 
From what I've read, I'd need to do the following:
git checkout fake-teams  
git rebase master
git checkout master
git merge fake-teams
Does that look right? If so, is this something I can do now, or have you already merged my changes so far?

**Comment by rohitpaulk** [comment]
@rorepo - You don't have to merge into master yourself. Commit your changes in your feature branch, and push them up. So basically, your workflow would be - 
```bash
$ git checkout fake-teams # You're now on a feature branch.

# Make changes, add , commit etc. 

# Once you're done, let's make sure that we're on the latest master before pushing up the branch.

$ git fetch upstream # Updates the local upstream/master to match the remote one.

$ git rebase upstream/master # This replays your commits on top of the latest master.

$ git push origin fake-teams # You can now create a PR using this branch.
```

Here, `origin` is the remote that points to `rorepo/gratipay.com` and `upstream` points to `gratipay/gratipay.com`

Additional reading: http://git-scm.com/book/ch3-6.html

**Comment by rorepo** [comment]
I've created a new branch 'fake-data', both to fix the 'noisy commit' as well as because the updates have gone beyond just teams. Will raise a new PR from that.

@whit537 - I tried removing tips, but looks like this will need a bit of figuring out. Will take that up next.

**Comment by rohitpaulk** [comment]
New PR - https://github.com/gratipay/gratipay.com/pull/3582

**Comment by whit537** [comment]
Ftr, noisy commits such as 1f7b01a99a6d747c2cd2a048499bb2b5e9f41932 result from running `git merge upstream/master` instead of `git rebase upstream/master` in @rohitpaulk's explanation.

> I tried removing tips, but looks like this will need a bit of figuring out. Will take that up next.

Cool, thanks. One question: do we want to leave fake `tips` around so that we can test things like migrating `tips` to `subscriptions`, and possibly UI we could/should add around informing users of changes to their giving ("you used to give to ___, and now subscribe to ___")? Thoughts @rohitpaulk @rorepo et al.?

**Pull request closed without merging**